test/opt/ccp_test.cpp

```diff
@@ -415,6 +415,55 @@ TEST_F(CCPTest, HandleAbortInstructions) {
   SinglePassRunAndMatch<opt::CCPPass>(spv_asm, true);
 }
 
+TEST_F(CCPTest, SSAWebCycles) {
+  // Test reduced from https://github.com/KhronosGroup/SPIRV-Tools/issues/1159
+  // When there is a cycle in the SSA def-use web, the propagator was getting
+  // into an infinite loop.  SSA edges for Phi instructions should not be
+  // added to the edges to simulate.
+  const std::string spv_asm = R"(
+               OpCapability Shader
+          %1 = OpExtInstImport "GLSL.std.450"
+               OpMemoryModel Logical GLSL450
+               OpEntryPoint Fragment %main "main"
+               OpExecutionMode %main OriginUpperLeft
+               OpSource GLSL 450
+               OpName %main "main"
+       %void = OpTypeVoid
+          %3 = OpTypeFunction %void
+        %int = OpTypeInt 32 1
+%_ptr_Function_int = OpTypePointer Function %int
+      %int_0 = OpConstant %int 0
+      %int_4 = OpConstant %int 4
+       %bool = OpTypeBool
+      %int_1 = OpConstant %int 1
+%_ptr_Output_int = OpTypePointer Output %int
+       %main = OpFunction %void None %3
+          %5 = OpLabel
+               OpBranch %11
+         %11 = OpLabel
+         %29 = OpPhi %int %int_0 %5 %22 %14
+         %30 = OpPhi %int %int_0 %5 %25 %14
+               OpLoopMerge %13 %14 None
+               OpBranch %15
+         %15 = OpLabel
+         %19 = OpSLessThan %bool %30 %int_4
+; CHECK: OpBranchConditional %true {{%\d+}} {{%\d+}}
+               OpBranchConditional %19 %12 %13
+         %12 = OpLabel
+; CHECK: OpIAdd %int %int_0 %int_0
+         %22 = OpIAdd %int %29 %30
+               OpBranch %14
+         %14 = OpLabel
+; CHECK: OpPhi %int %int_0 {{%\d+}} %int_0 {{%\d+}}
```

Comment:
This suggests an optimization:  If all value operands of a Phi are the same value %x then replace uses of the phi with %x.   I'll add that to #896

Reply:
That's already done, actually.  But in this case, we never get to it because we are purposely not adding SSA edges out of a Phi.  If the value of that Phi is used in a non-phi expression dominated by the Phi, it will be replaced with 0 (because the meet operator done in VisitPhi will return 0 as the meet value).

Reply:
We test for this in other tests in test_ccp.cpp

Reply:
I thought I had seen that somehwere but couldn't find it.  I didn't look far enough.

So to summarize (for myself):
- constant prop will get rid of the the non-phi dominated uses
- we could end up with these things as dead, and then dead code elim will get rid of most of them
- what remains is phis in cycles like this. 

That seems harmless/rare enough that it would be low priority to fix.